lib/web_of_science/queries.rb

```diff
@@ -13,7 +13,15 @@ def retrieve_by_id(uids)
     end
 
     # Convenience method, does the params_for_search expansion
-    # @param query [String] Query string like 'TS=particle swarm AND PY=(2007 OR 2008)'
+    # @param query_string [String] Query string like 'TS=particle swarm AND PY=(2007 OR 2008)'
+    # @return [WebOfScience::UidsQueryRestRetriever]
+    def user_query_uids(query_string, query_params: nil, batch_size: WebOfScience::BaseRestRetriever::MAX_RECORDS)
+      query = WebOfScience::UidsQueryRestRetriever::Query.new(user_query: query_string)
+      WebOfScience::UidsQueryRestRetriever.new(query, query_params:, batch_size:)
+    end
+
+    # Convenience method, does the params_for_search expansion
```

Comment:
our new method to query just for uids (we will use it from `QueryName` and `QueryOrcid`)